operator/internal/controller/auto/update_controller.go

```diff
@@ -612,6 +634,48 @@ func (s streamReader[T]) Result() (result, error) {
 	return res, fmt.Errorf("didn't receive a result")
 }
 
+// setStatusFromGRPCErr sets the Update object status blocks based on the result of a gRPC error.
+// If the error contains a structured PulumiErrorInfo, we'll use that information to surface the error.
+func (s streamReader[T]) setStatusFromGRPCErr(grpcErr error) error {
```

Comment:
It seems like this function doesn't really seek to change the error that is returned to the controller, and I'm not sure it should anyway. It seems to mutate the status, similarly to the block in the "Update complete" case.  So, maybe `setStatusFromGRPCErr` should not have a return value, for clarity.  It is purely a procedure.

Reply:
Ah, yes, we should just return the high-level grpc error anyway, since it'll contain more information.

Reply:
Updated to only update the status block.